test/libsolidity/smtCheckerTests/loops/do_while_bmc_iterations_break_8.sol

```diff
@@ -0,0 +1,19 @@
+contract C
+{
+	function f() public pure {
+		uint x;
+		do {
+			break;
+		} while (++x < 2);
+        // loop condition is not executed after break
+		assert(x == 0);
+	}
+}
+// ====
+// SMTEngine: bmc
+// SMTSolvers: z3
+// BMCLoopIterations: 3
+// ----
+//  Warning 5740: (79-86): Unreachable code.
+// Warning 2661: (79-82): BMC: Overflow (resulting value larger than 2**256 - 1) happens here.
```

Comment:
This does not feel right. Overflow should probably not be reported here, no?

Reply:
I opened an issue: https://github.com/ethereum/solidity/issues/14211 @leonardoalt @blishko

Reply:
This also affects loops, but maybe let's handle it, when solving the issue I opened.

Reply:
fixed by Leo in a separate PR

Reply:
We also decided to skip checks for constant loop conditions.
On one hand we had to reset variables used in a loop condition to do constant condition check. That's because value of a variable might be changed inside the loop body.
After the reset, increment expressions in the loop condition report overflow.